src/cpu/testers/dr_trace_player/trace_player.cc

```diff
@@ -167,24 +167,53 @@ DRTracePlayer::executeMemInst(DRTraceReader::TraceRef &mem_ref)
 bool
 DRTracePlayer::trySendMemRef(DRTraceReader::TraceRef &mem_ref)
 {
-    PacketPtr pkt = getPacket(mem_ref);
+    PacketPtr pkt, split_pkt;
+
+    std::tie(pkt, split_pkt) = getPacket(mem_ref);
+
+    if (!retrySplitPkt) {
+        // we should not be sending the first pkt again if
+        // we are retrying only on the second part of the
+        // previously stalled request
+        DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+
+        if (!port.sendTimingReq(pkt)) {
+            DPRINTF(DRTrace, "Failed to send pkt\n");
+            if (stats.memStallStart == 0) {
+                stats.memStallStart = curTick();
+            }
+            delete pkt;
+            if (!split_pkt) {
+            // if this is not a split pkt req,
+            // we can return back here
+            return true;
+            }
+        } else {
+            stats.latencyTracker[pkt] = curTick();
+            return false;
+        }
+    }
 
-    DPRINTF(DRTrace, "Trying to send %s\n", pkt->print());
+    DPRINTF(DRTrace, "Trying to send split %s\n", split_pkt->print());
 
-    if (!port.sendTimingReq(pkt)) {
-        DPRINTF(DRTrace, "Failed to send pkt\n");
+    if (!port.sendTimingReq(split_pkt)) {
+        DPRINTF(DRTrace, "Failed to send pkt (split pkt) \n");
         if (stats.memStallStart == 0) {
             stats.memStallStart = curTick();
         }
-        delete pkt;
+        retrySplitPkt = false;
+        delete split_pkt;
         return true;
     } else {
-        stats.latencyTracker[pkt] = curTick();
+        stats.latencyTracker[split_pkt] = curTick();
+        // also remember that we only need to retry on second part of
+        // the split pkt
+        retrySplitPkt = true;
```

Comment:
I think this is backwards. The `true` should go in the if statement above when the `sendTimingReq` fails